test/libp2p/eip-8.cpp

```diff
@@ -0,0 +1,336 @@
+		"c789a13bf5af6c0cb18515d0741e9a433d37838da86067f6069df56b282dba63ddfdada72d8ef3c4"
+		"dd2041b01deb463b9044e2a40a9c3ea79426229d0f3d901587fd706d578d7be87b7e8fbefce38b69"
+		"2cce17ea20017f512eb76b51a148dc21bb546fc8c8629e2d68c1b80e70b6082a5923911dd064c966"
+		"80a71c8f25997068b709f1cbd399a1"
+	);
+	auto handshake = TestHandshake::runWithInput(keyB, auth);
+	BOOST_REQUIRE(handshake->completedKeyEstablishment());
+	handshake->checkAuthValuesEIP8(56);
+}
+
+BOOST_AUTO_TEST_CASE(test_handshake_eip8_ack_plain)
+{
+	auto keyA = Secret("49a7b37aa6f6645917e7b807e9d1c00d4fa71f18343b0d4122a4d2df64dd6fee");
+	auto ack = fromHex(
+		"049f8abcfa9c0dc65b982e98af921bc0ba6e4243169348a236abe9df5f93aa69d99cadddaa387662"
+		"b0ff2c08e9006d5a11a278b1b3331e5aaabf0a32f01281b6f4ede0e09a2d5f585b26513cb794d963"
+		"5a57563921c04a9090b4f14ee42be1a5461049af4ea7a7f49bf4c97a352d39c8d02ee4acc416388c"
+		"1c66cec761d2bc1c72da6ba143477f049c9d2dde846c252c111b904f630ac98e51609b3b1f58168d"
+		"dca6505b7196532e5f85b259a20c45e1979491683fee108e9660edbf38f3add489ae73e3dda2c71b"
+		"d1497113d5c755e942d1"
+	);
+	auto initiatorPubk = NodeID("fda1cff674c90c9a197539fe3dfb53086ace64f83ed7c6eabec741f7f381cc803e52ab2cd55d5569bce4347107a310dfd5f88a010cd2ffd1005ca406f1842877");
+	auto handshake = TestHandshake::runWithInput(keyA, ack, initiatorPubk);
+	BOOST_REQUIRE(handshake->completedKeyEstablishment());
+	handshake->checkAckValuesEIP8(4);
+}
+
+BOOST_AUTO_TEST_CASE(test_handshake_eip8_ack1)
+{
+	auto keyA = Secret("49a7b37aa6f6645917e7b807e9d1c00d4fa71f18343b0d4122a4d2df64dd6fee");
+	auto ack = fromHex(
+		"019d04f43c232e6061efaa5fadb223e879eddce664706ff1764c6fe76dd482a9599a67dc5ae9a8fb"
+		"28eca063148ec619b227e0f70f676931d9a6ad1cd63c14bb611c13e9ff79957a93e13e1e509c2f12"
+		"1a3d69734fad4d1e6b5ff10d2d7d9b49d5beb5aff00f46ef4f20229bf9fb576203f418324e8724ba"
+		"b4190c99fe7912061a2c4cc1eb12221465a51ba753244eeeaf3fc789c3fc45cfc4674407f6b13f21"
+		"38a00b7a920f5c9bac8a6a01fc0b72ed3a59033c6fe855199b0ecf5f6abaea4ccc9da41a955c6f6e"
+		"8fcff6e7792033f7c0905e06ee1a47cb0cf88e00fdef70c1e217125f98cb3ef2c8f66cb8e940ed81"
+		"f7334c5a574c56ec3ed099cd98b5a2ba871ccbc819320b556709d199b382620d8b71f84234fe33b7"
+		"c95957bbbd4c1390a281502b8bf2ddf2e92631340b5aaeafb963fc62c96b2e98e93987bc197b2cfe"
+		"c08cdfe1aeb1da45fca5676e89725e897382e27ab2ae35de3e4733eaa9e76d9ca518ba074a9adf64"
+		"a28736a111137d346ca7743a5e0f4e7ab763d4331536ba8ef48163dc1dd5acede3a4b9b22ee31d3d"
+		"ac41e1f6cbfc78523449b3792b6523"
+	);
+	auto initiatorPubk = NodeID("fda1cff674c90c9a197539fe3dfb53086ace64f83ed7c6eabec741f7f381cc803e52ab2cd55d5569bce4347107a310dfd5f88a010cd2ffd1005ca406f1842877");
+	auto handshake = TestHandshake::runWithInput(keyA, ack, initiatorPubk);
+	BOOST_REQUIRE(handshake->completedKeyEstablishment());
+	handshake->checkAckValuesEIP8(4);
+}
+
+BOOST_AUTO_TEST_CASE(test_handshake_eip8_ack2)
+{
+	auto keyA = Secret("49a7b37aa6f6645917e7b807e9d1c00d4fa71f18343b0d4122a4d2df64dd6fee");
```

Comment:
I would prefer `Secret keyA("49...fee")` (similar below).
`auto` should be avoided for safety reasons (and for reasons of usability of IDEs) unless the type is some complicated template type.